### DISCUSSION TITLE
How to find error in jq

### DISCUSSION THREAD

**Discussion by MGoyal01** [stedolan/jq#3029] [Q&A]
I am getting trouble in finding error in jq. Can anyone guide me regarding that!!

**Comment by emanuele6** [comment]
Maybe?

**Reply by MGoyal01** [reply]
<img width="799" alt="Screenshot 2024-02-02 at 3 59 20 PM" src="https://github.com/jqlang/jq/assets/154582106/cfadec12-8a59-4b5b-9c8c-361337b3794a">

how to check at which line or which attribute is causing the error

**Reply by MGoyal01** [reply]
<img width="554" alt="Screenshot 2024-02-02 at 4 27 45 PM" src="https://github.com/jqlang/jq/assets/154582106/b925fe6a-6b9c-4f45-96b0-ed46668c76f6">

I  have tried  gojq

**Reply by MGoyal01** [reply]
the input can be thousands of line of data. 
so how to find at which line I am getting the error

**Reply by emanuele6** [reply]
@MGoyal01
Oh, you are asking how to find the line of input that contains the input that caused a `jq` error to occur; I completely misunderstood what you mean. :|

You could do that with something like:
```jq
try ["VALID", { sd: .user, ds: .user.officer[.user.jack] }] catch ["ERROR", input_filename, input_line_number, .]
```
That will output `["VALID",<resultsofyourexpression>...]` if it did not error; and `["ERROR", <filename>, <linenumber>, <errormessage>]` if it errored; in that case you would get:
```json
["ERROR","<stdin>",11,"Cannot index object with null"]
```

Note that it outputs `11` because that is the line on which the input that triggered the error ends.

**Reply by MGoyal01** [reply]
what does this input_line_number depicts? as the input data is about 8-9 lines not greater then 11

**Reply by wader** [reply]
@MGoyal01 i `input_line_number` will be the last line in the input that the current input value was part of. In the example above i guess 11 is the ending `}` of the single input value. Example with multiple input values:
```sh
$ jq -c '{line: input_line_number, value: .}' << EOF
123 {"a":1,
"b":2} "hello"
EOF
{"line":1,"value":123}
{"line":2,"value":{"a":1,"b":2}}
{"line":2,"value":"hello"}
```

Could you elaborate a bit more about how your input looks like? maybe a bit more detailed example input JSON?
